builtin/logical/pki/cert_util.go

```diff
@@ -134,6 +126,25 @@ func fetchCAInfo(ctx context.Context, b *backend, req *logical.Request, issuerRe
 	return caInfo, nil
 }
 
+// fetchCertBundle is our flex point to load either the legacy ca bundle if migration has yet to be
+// performed or load the bundle from the new key/issuer storage. Any function that needs a bundle
+// should load it using this method to maintain compatibility on secondary nodes for which their
+// primary's have not upgraded yet.
+func fetchCertBundle(ctx context.Context, b *backend, s logical.Storage, issuerRef string) (*certutil.CertBundle, error) {
+	if b.useLegacyBundleCaStorage() {
+		// We have not completed the migration so attempt to load the bundle from the legacy location
+		return getLegacyCertBundle(ctx, s)
+	}
+
+	id, err := resolveIssuerReference(ctx, s, issuerRef)
+	if err != nil {
+		// Usually a bad label from the user or misconfigured default.
+		return nil, err
```

Comment:
Not to nitpick, but doesn't this turn this into an `InternalError` versus a `UserError`? Not sure how much the distinction matters, but seems like we'd want to preserve that.

Reply:
Yup good catch, that was a flub on my part.